gu-hub-api.yaml

```diff
@@ -0,0 +1,582 @@
+        - application/json
+      parameters:
+        - name: changes
+          in: body
+          schema:
+            $ref: '#/definitions/HubSessionCommand'
+
+      responses:
+        200:
+          description: OK
+
+    delete:
+      tags:
+        - hub-session
+      responses:
+        200:
+          description: Deleted
+        404:
+          description: 'Session not found'
+  '/sessions/{sessionId}/config':
+    parameters:
+      - $ref: '#/parameters/sessionId'
+    get:
+      tags:
+        - hub-session
+      summary: Gets configuration from stash
+      operationId: getHubSessionConfig
+      responses:
+        '200':
+          description: OK
+          schema:
+            $ref: '#/definitions/ConfigStash'
+        '404':
+          description: Session not found
+        '500':
+          description: Internal Error
+    put:
+      tags:
+        - hub-session
+      summary: Sets configuration stash
+      operationId: setHubSessionConfiguration
+      parameters:
+        - name: config
+          description: New config stash value
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/ConfigStash'
+      responses:
+        '200':
+          description: OK
+          schema:
+            type: object
+            properties:
+              version:
+                type: integer
+            additionalProperties: true
+        '404':
+          description: Session not found
+        '409':
+          description: Conflicting version
+  '/sessions/{sessionId}/peer':
+    parameters:
+      - $ref: '#/parameters/sessionId'
+    post:
+      tags:
+        - hub-session
+      summary: Manually adds peers to hub session
+      operationId: addHubSessionPeers
+      parameters:
+        - name: peer_list
+          in: body
+          required: true
+          schema:
+            type: array
+            items:
+              type: string
+            uniqueItems: true
+      responses:
+        '200':
+          description: OK - peers added
+          schema:
+            type: array
+            items:
+              type: string
+              description: peer id
+            example:
+              - '0xb6804992598b9260a5b4ad553f98c3851cf2d18e'
+            description: list of session nodes after add
+        '404':
+          description: Session not found
+        '500':
+          description: Internal Error
+  '/sessions/{sessionId}/peer/{nodeId}/peer-sessions':
+    post:
+      tags:
+        - peer-session
+      operationId: createPeerSession
+      parameters:
+        - $ref: '#/parameters/sessionId'
+        - $ref: '#/parameters/nodeId'
+        - name: sessionSpec
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/PeerSessionSpec'
+      responses:
+        '201':
+          description: created
+          headers:
+            location:
+              type: string
+              format: uri
```

Comment:
there is no URI format in OpenAPI 2.0
```suggestion
```

Reply:
so what ? spec says:

"Primitives have an optional modifier property format. Swagger uses several known formats to more finely define the data type being used. However, the format property is an open string-valued property, and can have any value to support documentation needs. Formats such as "email", "uuid", etc., can be used even though they are not defined by this specification. "